misk-jdbc/src/main/kotlin/misk/jdbc/Transacter.kt

```diff
@@ -14,19 +14,41 @@ interface Transacter {
    * If the work raises an exception the transaction will be rolled back instead of committed.
    *
    * It is an error to start a transaction if another transaction is already in progress.
+   *
+   * Prefer using [transactionWithSession] instead of this method as it has more functionality such
+   * as commit hooks.
    */
+  @Deprecated("Use transactionWithSession instead",
+    replaceWith = ReplaceWith("transactionWithSession(work)")
+  )
   fun <T> transaction(work: (connection: Connection) -> T): T
+
+  /**
+   * Starts a transaction on the current thread, executes [work], and commits the transaction.
+   * If the work raises an exception the transaction will be rolled back instead of committed.
+   *
+   * This session object passed in wraps a connection and provides a way to add pre and post commit
+   * hooks that execute before and after a transaction is committed.
+   *
+   * It is an error to start a transaction if another transaction is already in progress.
+   */
+  fun <T> transactionWithSession(work: (session: JDBCSession) -> T): T
```

Comment:
I ideally wanted to call this transaction() as well, but kotlin java interop doesn't work well with the same function name.